app/javascript/components/AddPopup/AddPopup.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { has } from 'ramda';
+
+import Button from '@material-ui/core/Button';
+import Card from '@material-ui/core/Card';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import CardHeader from '@material-ui/core/CardHeader';
+import CloseIcon from '@material-ui/icons/Close';
+import IconButton from '@material-ui/core/IconButton';
+import Modal from '@material-ui/core/Modal';
+import TextField from '@material-ui/core/TextField';
+
+import TaskForm from 'forms/TaskForm';
+
+import useStyles from './useStyles';
+
+const AddPopup = ({ onClose, onCreateCard }) => {
+  const [task, changeTask] = useState(TaskForm.defaultAttributes());
+  const [isSaving, setSaving] = useState(false);
+  const [errors, setErrors] = useState({});
+  const handleCreate = () => {
+    setSaving(true);
+
+    onCreateCard(task).catch((error) => {
+      setSaving(false);
+      setErrors(error || {});
+
+      if (error instanceof Error) {
+        alert(`Creation Failed! Error: ${error.message}`);
+      }
+    });
+  };
+  const handleChangeTextField = (fieldName) => (event) => changeTask({ ...task, [fieldName]: event.target.value });
```

Comment:
You can use destructuring:
`const handleChangeTextField = fieldName => ({ target: { value }) => changeTask({ ...task, [fieldName]: value });`